### PR TITLE
WIP ref(worker): switch to @azure/brigadier

### PR DESCRIPTION
**Pull request by technosophos**
This refactors the Brigade internal library to use @Azure/brigadier. This contains breaking changes in the pathing. (See https://www.npmjs.com/package/@azure/brigadier)

This also intercepts and overrides `require("@azure/brigadier")` for brigade.js files.

### CODE REVIEW

**Comment by technosophos** [comment]
Still todo:

- [ ] Manual testing
- [ ] Publish Azure/brigadier GitHub project

File: brigade-worker/src/app.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /**
  * Module app is the main application runner.
  */
-import * as events from "./events";
+import * as events from "@azure/brigadier/out/events";
```
Comment (radu-matei) [inline brigade-worker/src/app.ts]: nit: I am mildly annoyed by the fact that `/out/` should be included every time the package is used - I suspect there's a configuration property that can help us avoid this - I'll look into it tomorrow morning.

Comment (radu-matei) [reply brigade-worker/src/app.ts]: Ok, mildly was an understatement.

File: brigade-worker/src/k8s.ts [hunk 2]
```diff
@@ -238,8 +238,12 @@ export class JobRunner implements jobs.JobRunner {
   pod: kubernetes.V1Pod;
   cancel: boolean;
   reconnect: boolean;
+  
+  constructor() {}
 
-  constructor(job: jobs.Job, e: BrigadeEvent, project: Project) {
+  public init<T extends jobs.Job>(job: T, e: BrigadeEvent, project: Project) {
+    // init takes a generic so we can run this against mocks as well as against the real
```
Comment (radu-matei) [inline brigade-worker/src/k8s.ts]: nit: can the function comment be before the function definition?

File: brigade-worker/src/app.ts (hunk 1 above)
Comment (radu-matei) [reply brigade-worker/src/app.ts]: Update: found a solution for pretty import paths - not ideal, but will PR against Brigadier.

File: brigade-worker/src/k8s.ts (hunk 2 above)
Comment (technosophos) [reply brigade-worker/src/k8s.ts]: fixed

**Pull request merged**